### ISSUE TITLE
Polygon() crashed

### ISSUE THREAD

**Issue by zhangliye** [JuliaGeo/LibGEOS.jl#29]
pts = [[0.,0.],[10.,0.], [10.,10.],[0.,10.] ]
Polygon([pts])    # crashed here

pts = [[0.,0.],[10.,0.], [10.,10.],[0.,10.] ]
Polygon([pts])    # correct

suggestion:
(1) Polygon()  should check the input and add the end point if last input point is not same as the first point.
(2) throw an error rather than crash. 

# running on Win7, Julia 5.0, LibGEOS v0.1.0

**Comment by yeesian** [comment]
Sounds reasonable. Will you like to try a PR for this?

**Comment by zhangliye** [comment]
I am not sure how to handle PR. I will try.

**Comment by yeesian** [comment]
Thanks!

**Comment by yeesian** [comment]
Hey Liye, I haven't been able to reproduce this issue:

```julia
julia> using LibGEOS

julia> pts = [[0.,0.],[10.,0.], [10.,10.],[0.,10.] ]
4-element Array{Array{Float64,1},1}:
 [0.0,0.0]
 [10.0,0.0]
 [10.0,10.0]
 [0.0,10.0]

julia> Polygon([pts]) # crashed here
ERROR: LibGEOS: Error in GEOSGeom_createPolygon
 in createPolygon(::Ptr{Ptr{Void}}, ::Array{Ptr{Ptr{Void}},1}) at /Users/yeesian/.julia/v0.5/LibGEOS/src/geos_functions.jl:371
 in LibGEOS.Polygon(::Array{Array{Array{Float64,1},1},1}) at /Users/yeesian/.julia/v0.5/LibGEOS/src/geos_types.jl:74

julia> pts = [[0.,0.],[10.,0.], [10.,10.],[0.,10.],[0.,0.]]
5-element Array{Array{Float64,1},1}:
 [0.0,0.0]
 [10.0,0.0]
 [10.0,10.0]
 [0.0,10.0]
 [0.0,0.0]

julia> Polygon([pts]) # correct
LibGEOS.Polygon(Ptr{Ptr{Void}} @0x00007fbbdc559dd8)
```

Admittedly, we can make [the error message](https://github.com/JuliaGeo/LibGEOS.jl/blob/929a4fa24b9e3ba70090fa7523471a8f2788ff0f/src/geos_functions.jl#L371) more informative, and suggest that it could be due to the rings not being closed. But we do throw an error, rather than crashing julia.

Is it still an issue for you?

**Comment by visr** [comment]
On windows this is still an issue, may be related to #35
```julia
julia> using LibGEOS
julia> pts = [[0.,0.],[10.,0.], [10.,10.],[0.,10.] ];
julia> Polygon([pts])
```
```
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x0 -- unknown function (ip: 0000000000000000)
while loading no file, in expression starting on line 0
unknown function (ip: FFFFFFFFFFFFFFFF)
GEOSGeom_createLinearRing_r at C:\Users\visser_mn\.julia\v0.6\WinRPM\deps\usr\x86_64-w64-mingw32\sys-root\mingw\bin\libgeos_c-1.DLL (unknown line)
Type at C:\Users\visser_mn\.julia\v0.6\LibGEOS\src\geos_types.jl:72
```

**Comment by yeesian** [comment]
can you check and see if this happens with the latest binaries in #37?

**Comment by evetion** [comment]
Added testcase for this in https://github.com/JuliaGeo/LibGEOS.jl/pull/38.

**Comment by yeesian** [comment]
closed by #38
